tests/encodeinput.cpp

```diff
@@ -167,40 +167,49 @@ EncodeOutput::~EncodeOutput()
         fclose(m_fp);
 }
 
-EncodeOutput* EncodeOutput::create(const char* outputFileName, int width , int height)
+EncodeOutput* EncodeOutput::create(const char* outputFileName, int width,
+                                   int height, const char* codecName)
 {
-    EncodeOutput * output = NULL;
-    if(outputFileName==NULL)
+    EncodeOutput* output = NULL;
+    if (outputFileName == NULL)
         return NULL;
-    const char *ext = strrchr(outputFileName,'.');
-    if(ext==NULL)
+    const char* ext = strrchr(outputFileName, '.');
+    std::string codec(""); // needed for strcasecmp to work
+    if (!ext && !codecName)
         return NULL;
-    ext++;//h264;264;jsv;avc;26l;jvt;ivf
-    if(strcasecmp(ext,"h264")==0 ||
-        strcasecmp(ext,"264")==0 ||
-        strcasecmp(ext,"jsv")==0 ||
-        strcasecmp(ext,"avc")==0 ||
-        strcasecmp(ext,"26l")==0 ||
-        strcasecmp(ext,"jvt")==0 ) {
-            output = new EncodeOutputH264();
+    ext++;
+
+    if (codecName)
+        codec.assign(codecName);
+
+    if (!strcasecmp("AVC", codec.c_str()) || !strcasecmp(ext, "h264")
+        || !strcasecmp(ext, "264") || !strcasecmp(ext, "jsv")
+        || !strcasecmp(ext, "avc") || !strcasecmp(ext, "26l")
+        || !strcasecmp(ext, "jvt")) {
+        output = new EncodeOutputH264();
+    }
+    else if (!strcasecmp("VP8", codec.c_str())
+             || (!strcasecmp("VP8", codec.c_str()) && !strcasecmp(ext, "ivf"))
```

Comment:
@chivakker , I agree with you, we don't need to suppose vp8 codec type for ivf, it will force a vp8 encoding but maybe user want a vp9 encoder. I recommend we use xxx_vp9.ivf or xxx_vp8.ivf, it should be better than xxx.vp8/xxx.vp9, becasue we can know it is ivf format not a pure vp8/vp9 bitstream.